poetry/installation/executor.py

```diff
@@ -109,49 +101,54 @@ def execute(self, operations):
 
         # We group operations by priority
         groups = itertools.groupby(operations, key=lambda o: -o.priority)
-        i = 0
         self._sections = OrderedDict()
         for _, group in groups:
-            for chunk in chunked(group, self._max_workers):
-                tasks = []
-                for operation in chunk:
-                    if self.supports_fancy_output():
-                        if id(operation) not in self._sections:
-                            if self._should_write_operation(operation):
-                                self._lock.acquire()
-                                self._sections[id(operation)] = self._io.section()
-                                self._sections[id(operation)].write_line(
-                                    "  <fg=blue;options=bold>•</> {message}: <fg=blue>Pending...</>".format(
-                                        message=self.get_operation_message(operation),
-                                    ),
-                                )
-                                self._lock.release()
-                    else:
+            tasks = []
+            for operation in group:
+                if self._shutdown:
+                    break
+
+                if self.supports_fancy_output():
+                    if id(operation) not in self._sections:
                         if self._should_write_operation(operation):
-                            if not operation.skipped:
-                                self._io.write_line(
-                                    "  <fg=blue;options=bold>•</> {message}".format(
-                                        message=self.get_operation_message(operation),
-                                    ),
-                                )
-                            else:
-                                self._io.write_line(
-                                    "  <fg=default;options=bold,dark>•</> {message}: "
-                                    "<fg=default;options=bold,dark>Skipped</> "
-                                    "<fg=default;options=dark>for the following reason:</> "
-                                    "<fg=default;options=bold,dark>{reason}</>".format(
-                                        message=self.get_operation_message(operation),
-                                        reason=operation.skip_reason,
-                                    )
+                            self._lock.acquire()
+                            self._sections[id(operation)] = self._io.section()
+                            self._sections[id(operation)].write_line(
```

Comment:
i still think the `Pending` messages need to move into the task executor since (unless i'm misreading) otherwise _every_ package gets marked as pending (which while true fills the screen pushing the stuff actually being installed off-screen)

maybe we don't need the `Pending` markers at all?

Reply:
It's changed so that the pending message is only displayed in the `_do_execute_operation()` method.